qualtran/resource_counting/_algorithm_summary_counts.py

```diff
@@ -0,0 +1,86 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import logging
+from typing import Callable, Dict, TYPE_CHECKING
+
+from attrs import frozen
+
+from ._bloq_counts import bloq_is_clifford
+from ._call_graph import get_bloq_callee_counts
+from ._costing import CostKey
+
+if TYPE_CHECKING:
+    from qualtran import Bloq
+
+AlgorithmSummaryDict = Dict[str, int]
+
+logger = logging.getLogger(__name__)
+
+
+@frozen
+class AlgorithmSummaryCounts(CostKey[AlgorithmSummaryDict]):
```

Comment:
Isn't this mostly redundant with the existing `QECGatesCost` cost key which uses the `GateCounts` data class? 

The `GateCounts` numbers are all factorized, so you should be able to add up the ones that need adding up and create an `AlgorithmsSummary`, no? If not: what are the logical incompatibilities?

Reply:
rotations ... the QECGatesCost ignores rotations (i.e. gives a GateCounts of all zeros)

Reply:
can we add that then

Reply:
done